src/corehost/common/pal.windows.cpp

```diff
@@ -54,6 +54,24 @@ bool pal::find_coreclr(pal::string_t* recv)
     return false;
 }
 
+void pal::setup_api_sets(const std::unordered_set<pal::string_t>& api_sets)
+{
+    pal::string_t path;
+
+    (void) getenv(_X("PATH"), &path);
+
+    // We need this ugly hack, as the PInvoked DLL's static dependencies can come from
+    // some other NATIVE_DLL_SEARCH_DIRECTORIES and not necessarily side by side. However,
+    // CoreCLR.dll loads PInvoke DLLs with LOAD_WITH_ALTERED_SEARCH_PATH. Note that this
+    // option cannot be combined with LOAD_LIBRARY_SEARCH_USER_DIRS, so the AddDllDirectory
+    // doesn't help much in telling CoreCLR where to load the PInvoke DLLs from.
+    // So we resort to modifying the PATH variable on our own hoping Windows loader will do the right thing.
+
+    trace::verbose(_X("Setting PATH=%s"), path.c_str());
+
+    // Assuming no other thread modified PATH between Get and SetEnvironmentVariableW.
```

Comment:
At this point in time, there should not be any other thread running since the host is prepping to start the runtime.